pkg/nsx/client.go

```diff
@@ -58,30 +66,55 @@ func GetClient(cf *config.NSXOperatorConfig) *Client {
 	vspherelog.SetLogger(logger)
 	c := NewConfig(strings.Join(cf.NsxApiManagers, ","), cf.NsxApiUser, cf.NsxApiPassword, "", 10, 3, 20, 20, true, true, true, ratelimiter.AIMD, cf.GetTokenProvider(), nil, cf.Thumbprint)
 	cluster, _ := NewCluster(c)
-	nsxVersion, err := cluster.GetVersion()
-	if err != nil {
-		log.Error(err, "get version error")
-		return nil
-	}
-	err = nsxVersion.Validate(minVersion)
-	if err != nil {
-		log.Error(err, "validate version error")
-		return nil
-	}
+
 	queryClient := search.NewQueryClient(restConnector(cluster))
 	groupClient := domains.NewGroupsClient(restConnector(cluster))
 	securityClient := domains.NewSecurityPoliciesClient(restConnector(cluster))
 	ruleClient := security_policies.NewRulesClient(restConnector(cluster))
 	nsxChecker := &NSXHealthChecker{
 		cluster: cluster,
 	}
-	return &Client{
+	nsxVersionChecker := &NSXVersionChecker{
+		cluster:                 cluster,
+		securityPolicySupported: false,
+	}
+
+	nsxClient := &Client{
 		NsxConfig:      cf,
 		RestConnector:  restConnector(cluster),
 		QueryClient:    queryClient,
 		GroupClient:    groupClient,
 		SecurityClient: securityClient,
 		RuleClient:     ruleClient,
 		NSXChecker:     *nsxChecker,
+		NSXVerChecker:  *nsxVersionChecker,
+	}
+	nsxClient.NSXCheckVersionForSecurityPolicy()
```

Comment:
Add comments why not exiting here, or print a warning log?

Reply:
Done